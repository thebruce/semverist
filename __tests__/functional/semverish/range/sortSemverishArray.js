

const rangeFactory = require('../../../../lib/semverish/range');

let tmpMocks = [];
let ranger;

describe('Sort Semverish Array tests', () => {
  beforeEach(() => {
    ranger = rangeFactory('semverist', 'range');
    tmpMocks.forEach(mock => mock.mockRestore());
    tmpMocks = [];
    jest.resetAllMocks();
    jest.spyOn(Date, 'now').mockReturnValue(2000);
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  test('sortSemverishArray', () => {
    expect.assertions(1);
    return ranger
      .then((RangeClass) => {
        const rangeClass = new RangeClass();
        return rangeClass.sortSemverishArray(
          [
            '2.0.0',
            '1.0.0',
          ]
        );
      })
      .then(obj => expect(obj).toEqual([
        '1.0.0',
        '2.0.0',
      ]));
  });

  test('sortSemverishArrayMixedSemverish', () => {
    expect.assertions(1);
    return ranger
      .then((RangeClass) => {
        const rangeClass = new RangeClass();
        return rangeClass.sortSemverishArray(
          [
            '1',
            '1.2',
            '1.1',
            '1.2.1',
          ]
        );
      })
      .then(obj => expect(obj).toEqual([
        '1',
        '1.1',
        '1.2',
        '1.2.1',
      ]));
  });

  test('sortSemverishArrayWayMixedSemverish', () => {
    expect.assertions(1);
    return ranger
      .then((RangeClass) => {
        const rangeClass = new RangeClass();
        return rangeClass.sortSemverishArray(
          [
            '1',
            '1.2',
            '1.1',
            '1.5.1',
            '1.2.1',
            '1.3',
          ]
        );
      })
      .then(obj => expect(obj).toEqual([
        '1',
        '1.1',
        '1.2',
        '1.2.1',
        '1.3',
        '1.5.1',
      ]));
  });

  test('sortSemverishArrayWayMixedSemverishWithEquivs', () => {
    expect.assertions(1);
    return ranger
      .then((RangeClass) => {
        const rangeClass = new RangeClass();
        return rangeClass.sortSemverishArray(
          [
            '1.0.0',
            '1.0',
            '1.1',
            '1.2',
            '1.1.0',
            '1.2.0',
            '1.3',
            '1.5.1',
            '1',
          ]
        );
      })
      .then(obj => expect(obj).toMatchSnapshot());
  });

  test('sortSemverishArrayNull', () => {
    expect.assertions(1);
    return rangeFactory('semverist', 'range')
      .then((RangeClass) => {
        const range = new RangeClass();
        return range.sortSemverishArray();
      })
      .catch((e) => {
        expect(e.message).toEqual('Can not sort a semverish Array that is not actually range.');
      });
  });

  test('sortSemverishArrayString', () => {
    expect.assertions(1);
    return rangeFactory('semverist', 'range')
      .then((RangeClass) => {
        const range = new RangeClass();
        return range.sortSemverishArray('not a range');
      })
      .catch((e) => {
        expect(e.message).toEqual('Can not sort a semverish Array that is not actually range.');
      });
  });
});
