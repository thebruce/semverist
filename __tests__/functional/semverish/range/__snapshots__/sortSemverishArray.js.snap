// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort Semverish Array tests sortSemverishArrayWayMixedSemverishWithEquivs 1`] = `
Array [
  "1",
  "1.0",
  "1.0.0",
  "1.1",
  "1.1.0",
  "1.2",
  "1.2.0",
  "1.3",
  "1.5.1",
]
`;
